sqldelight-compiler/src/main/kotlin/com/squareup/sqldelight/core/compiler/QueryGenerator.kt

```diff
@@ -60,6 +60,8 @@ abstract class QueryGenerator(private val query: BindableQuery) {
     // The number of non-array bindArg's we've encountered so far
     var nonArrayBindArgsCount = 0
 
+    val argumentNames = query.arguments.map { it.type.name }.toSet()
```

Comment:
switch this to `NameAllocator().apply { add all argument names }` and then you dont need the `getNonConflictingName method, you can just call `val elementName = nameAllocator.get(type.name)`